bugtool/configuration/cilium.go

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package configuration
+
+import (
+	dump "github.com/cilium/cilium/bugtool/dump"
+)
+
+func CiliumTasks() dump.Tasks {
+	ts := dump.Tasks{}
+	for _, cmd := range ciliumCommandsJSON() {
+		ts = append(ts, createExecFromString(cmd, "json"))
+	}
+	return ts
+}
+
+func ciliumCommandsJSON() []string {
+	var commands []string
+	generators := []func() []string{
+		CiliumAgentAPICLICommands,
+	}
+
+	for _, generator := range generators {
+		commands = append(commands, generator()...)
+	}
+
+	return commands
+}
+
+// CiliumAgentAPICLICommands returns a list of all dump commands that use the
+// Ciliums Agent CLI binary to dump data from the Agents API.
+func CiliumAgentAPICLICommands() []string {
+	commands := []string{
+		"cilium debuginfo --output=json", // debuginfo uses different output flag format.
+	}
+	// TODO: Many of these are redundant with debuginfo, clean them up.
+	for _, cmd := range []string{
+		"cilium metrics list",
+		"cilium fqdn cache list",
+		"cilium config -a",
+		"cilium encrypt status",
+		"cilium endpoint list",
+		"cilium bpf bandwidth list",
+		"cilium bpf tunnel list",
+		"cilium bpf lb list",
```

Comment:
The JSON output for this one seems weirdly broken.

Reply:
Its not very well crafted json but it does print valid json for me:

ex.
```
{
  "10.96.0.10:53": [
    "10.244.0.186:53 (2) (1)",
    "10.244.0.90:53 (2) (2)",
    "0.0.0.0:0 (2) (0) [ClusterIP, non-routable]"
  ],
  "10.96.0.10:9153": [
    "10.244.0.186:9153 (3) (1)",
    "0.0.0.0:0 (3) (0) [ClusterIP, non-routable]",
    "10.244.0.90:9153 (3) (2)"
  ],
  "10.96.0.1:443": [
    "0.0.0.0:0 (1) (0) [ClusterIP, non-routable]",
    "172.19.0.3:6443 (1) (1)"
  ]
}
```

Reply:
many of these use a generic dump-to-json function for these. I am planning to improve these in a separate PR.

Reply:
For now, we'll have the human readable format as well.